test-framework/junit5-mockito/src/main/java/io/quarkus/test/junit/mockito/InjectMock.java

```diff
@@ -12,4 +12,10 @@
 @Target(ElementType.FIELD)
 @Retention(RetentionPolicy.RUNTIME)
 public @interface InjectMock {
+
+    /**
+     * If true, then Quarkus will change the scope of {@code Singleton} beans to {@code ApplicationScoped}
+     * to make the mockable
```

Comment:
We might need a more detailed description and also mention that the conversion might result in unexpected behavior, e.g. it's OK to read `@Singleton` fields but it's not OK for client proxies...

Reply:
Done